lnwire/interfaces.go

```diff
@@ -11,3 +13,24 @@ type AnnounceSignatures interface {
 
 	Message
 }
+
+// ChannelAnnouncement is an interface that must be satisfied by any message
+// used to announce and prove the existence of a channel.
+type ChannelAnnouncement interface {
```

Comment:
Similarly, could also abstract validation of the chan ann behind this as well. Though at that point, maybe we'd move the interface somewhere else, or implement that as a superset interface.

Reply:
ok have done this now but another review comment suggests that lnwire should not contain validation logic... what do you guys think is the best approach here? 

should we let the interfaces only describe underlying fields and then have separate interfaces somewhere else (netann) that hides the validation / sig check logic?

Reply:
or perhaps this interface stays here but we instead just keep the Validation functions separate in `netann`?

Reply:
> or perhaps this interface stays here but we instead just keep the Validation functions separate in netann?

+1

Reply:
updated!